File: HospitalInformationSystem.Shared/Entities/Patient.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HospitalInformationSystem.Shared.Entities
{
    [Index(nameof(FileNo), IsUnique = true)]
    [Index(nameof(CitizenId), IsUnique = true)]
    public class Patient
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        [Required(ErrorMessage = "Name is required.")]
        [StringLength(255, ErrorMessage = "Name cannot exceed {1} characters.")]
        [DataType(DataType.Text)]
        public string Name { get; set; }

        [Required(ErrorMessage = "FileNo is required.")]
        [Column(TypeName = "decimal(38, 0)")]
        public decimal FileNo { get; set; }

        [Required(ErrorMessage = "Citizen Id is required.")]
        [StringLength(11, MinimumLength = 11, ErrorMessage = "Citizen ID must contain {1} digits.")]
        public string CitizenId { get; set; }

        [DisplayFormat(DataFormatString = "{0:d}")]
        [DataType(DataType.Date)]
        public DateTime? BirthDate { get; set; }

        [EnumDataType(typeof(Gender))]
        public Gender? Gender { get; set; }

        [StringLength(65, ErrorMessage = "Natinality cannot exceed {1} characters.")]
        [DataType(DataType.Text)]
        public string Natinality { get; set; }

        [Required(ErrorMessage = "Phone Number is required.")]
        [StringLength(15, ErrorMessage = "Phone Number cannot exceed {1} numbers.")]
        [DataType(DataType.PhoneNumber)]
        [Phone]
        public string PhoneNumber { get; set; }

        [StringLength(512, ErrorMessage = "Email Address cannot exceed {1} characters.")]
        [DataType(DataType.EmailAddress)]
        [EmailAddress]
        public string Email { get; set; }

        [StringLength(255, ErrorMessage = "Contact Person cannot exceed {1} characters.")]
        [DataType(DataType.Text)]
        public string ContactPerson { get; set; }

        [StringLength(255, ErrorMessage = "Contact Relation cannot exceed {1} characters.")]
        [DataType(DataType.Text)]
        public string ContactRelation { get; set; }

        [StringLength(15, ErrorMessage = "Phone Number cannot exceed {1} numbers.")]
        [DataType(DataType.PhoneNumber)]
        [Phone]
        public string ContactPhone { get; set; }

        [Required(ErrorMessage = "Country is required.")]
        [StringLength(65, ErrorMessage = "Country cannot exceed {1} characters.")]
        [DataType(DataType.Text)]
        public string Country { get; set; }

        [Required(ErrorMessage = "City is required.")]
        [StringLength(65, ErrorMessage = "City cannot exceed {1} characters.")]
        [DataType(DataType.Text)]
        public string City { get; set; }

        [StringLength(255, ErrorMessage = "Street cannot exceed {1} characters.")]
        [DataType(DataType.Text)]
        public string Street { get; set; }

        [StringLength(512, ErrorMessage = "Address1 cannot exceed {1} characters.")]
        [DataType(DataType.MultilineText)]
        public string Address1 { get; set; }

        [StringLength(512, ErrorMessage = "Address2 cannot exceed {1} characters.")]
        [DataType(DataType.MultilineText)]
        public string Address2 { get; set; }

        [DisplayFormat(DataFormatString = "{0:d}")]
        [DataType(DataType.Date)]
        public DateTime? FirstVisitDate { get; set; }

        [DisplayFormat(DataFormatString = "{0:d}")]
        [DataType(DataType.Date)]
        public DateTime RecordCreationDate { get; set; }
    }
}

File: HospitalInformationSystemUI/src/app/models/patient.ts
import { Guid } from "guid-typescript";

export class Patient {
    id: Guid;
    name: string;
    fileNo: number;
    citizenId: string;
    birthDate: Date;
    gender: 0 | 1;
    natinality: string;
    phoneNumber: string;
    email: string;
    contactPerson: string;
    contactRelation: string;
    contactPhone: string;
    country: string;
    city: string;
    street: string;
    address1: string;
    address2: string;
    firstVisitDate: Date;
    recordCreationDate: Date;

    constructor(name = null, fileNo = null, citizenId = null, birthDate = null, gender = null, natinality = null, phoneNumber = null, email = null, contactPerson = null, contactRelation = null, contactPhone = null, country = null, city = null, street = null, address1 = null, address2 = null, firstVisitDate = null) {
        this.name = name;
        this.fileNo = fileNo;
        this.citizenId = citizenId;
        this.birthDate = birthDate;
        this.gender = gender;
        this.natinality = natinality;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.contactPerson = contactPerson;
        this.contactRelation = contactRelation;
        this.contactPhone = contactPhone;
        this.country = country;
        this.city = city;
        this.street = street;
        this.address1 = address1;
        this.address2 = address2;
        this.firstVisitDate = firstVisitDate;
    }
}